src/lib.rs

```diff
@@ -439,6 +450,98 @@ impl DstLayout {
         }
     }
 
+    /// Extends `self` with the layout of `field`, adding whatever padding is
+    /// needed to satisfy the field's natural alignment or `max_align`
+    /// (whichever comes first). If `self` corresponds to a type marked with
+    /// `repr(packed(N))`, then `max_align` should be `Some(N)`, otherwise
+    /// `None`.
+    ///
+    /// This method does **not** add trailing padding needed to satisfy the
+    /// type's overall alignment requirements.
+    ///
+    ///
+    #[allow(dead_code)]
+    pub(crate) const fn extend(
+        self,
+        field: DstLayout,
+        packed_modifier: Option<NonZeroUsize>,
```

Comment:
Feel free to take or leave this; it strikes me that this is a bit more self-explanatory for programmers who are reasoning about this method in terms of its use in a derive.

```suggestion
        repr_packed: Option<NonZeroUsize>,
```

Alternatively, if you prefer this not be tied to its intended use, maybe this might be better:

```suggestion
        max_align: Option<NonZeroUsize>,
```